src/pages/openSourceApply.js

```diff
@@ -0,0 +1,170 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Flex, Box } from '@rebass/grid';
+
+import Page from '../components/Page';
+import Loading from '../components/Loading';
+import { withUser } from '../components/UserProvider';
+import withIntl from '../lib/withIntl';
+import { addCreateCollectiveFromGithubMutation } from '../graphql/mutations';
+import StyledCard from '../components/StyledCard';
+import { H3, P } from '../components/Text';
+import StyledButton from '../components/StyledButton';
+import GithubRepositories from '../components/GithubRepositories';
+import StyledInputField from '../components/StyledInputField';
+import MessageBox from '../components/MessageBox';
+import SignInOrJoinFree from '../components/SignInOrJoinFree';
+import { Router } from '../server/pages';
+
+import { getGithubRepos } from '../lib/api';
+
+const { WEBSITE_URL } = process.env;
+
+class OpenSourceApplyPage extends Component {
+  static async getInitialProps({ query }) {
+    return {
+      token: query && query.token,
+    };
+  }
+
+  static propTypes = {
+    token: PropTypes.string,
+    loadingLoggedInUser: PropTypes.bool,
+    LoggedInUser: PropTypes.object,
+    createCollectiveFromGithub: PropTypes.func,
+  };
+
+  state = {
+    result: {},
+    loadingRepos: false,
+    repositories: [],
+    creatingCollective: false,
+  };
+
+  async componentDidMount() {
+    const { token } = this.props;
+    if (!token) {
+      return;
+    }
+    this.setState({ loadingRepos: true });
+
+    try {
+      const repositories = await getGithubRepos(token);
+      if (repositories.length !== 0) {
+        this.setState({ repositories, loadingRepos: false, result: {} });
+      } else {
+        this.setState({
+          loadingRepos: false,
+          result: { type: 'info', mesg: 'Info: No Repository found' },
+        });
+      }
+    } catch (error) {
+      this.setState({
+        loadingRepos: false,
+        result: { type: 'error', mesg: 'Error: An unknown error occured' },
+      });
+    }
+  }
+
+  renderContent() {
+    const { token, LoggedInUser } = this.props;
+    const { repositories } = this.state;
+
+    if (!LoggedInUser) {
+      return <SignInOrJoinFree redirect={Router.asPath} />;
+    } else if (!token || repositories.length === 0) {
+      return this.renderConnectGithubButton();
+    } else {
+      return this.renderGithubRepos();
+    }
+  }
+
+  async createCollectives(collectiveInputType) {
+    collectiveInputType.type = 'COLLECTIVE';
+    try {
+      const res = await this.props.createCollectiveFromGithub(collectiveInputType);
+      const collective = res.data.createCollectiveFromGithub;
+      Router.pushRoute('collective', {
+        slug: collective.slug,
+        status: 'collectiveCreated',
+        CollectiveId: collective.id,
```

Comment:
Do we need `CollectiveId`?

Reply:
Actually looked into how it is used here https://github.com/opencollective/opencollective-frontend/blob/master/src/components/CreateCollective.js#L124

Reply:
Let's remove if we don't see the purpose (I personally don't see the purpose).

Reply:
Same for `CollectiveSlug`